plugins/acl/static-role-based/src/main/java/org/apache/cloudstack/acl/StaticRoleBasedAPIAccessChecker.java

```diff
@@ -65,43 +65,72 @@ public StaticRoleBasedAPIAccessChecker() {
         }
     }
 
+    /**
+     * Only one strategy should be used between StaticRoleBasedAPIAccessChecker and DynamicRoleBasedAPIAccessChecker
+     * Default behavior is to use the Dynamic version. The StaticRoleBasedAPIAccessChecker is the legacy version.
+     * If roleService is enabled, then it uses the DynamicRoleBasedAPIAccessChecker, otherwise, it will use the
+     * StaticRoleBasedAPIAccessChecker.
+     */
     @Override
     public boolean isEnabled() {
-        return !isDisabled();
-    }
-    public boolean isDisabled() {
+        if (roleService.isEnabled()) {
+            LOGGER.debug("RoleService is enabled. We will use it instead of StaticRoleBasedAPIAccessChecker.");
+        }
         return roleService.isEnabled();
```

Comment:
seems like the logic is reversed. was this on purpose?
it seems the `StaticRoleBasedAPIAccessChecker` shhould return false if `roleService.isEnabled() == true`